google-cloud-bigtable/test/google/cloud/bigtable/conformance_test.rb

```diff
@@ -51,29 +50,37 @@ def self.build_test_for test, index
                               app_profile_id: nil
                             ]
 
-      results = test.results.to_a
-      if results.map(&:error).include? true
+      rows = table.read_rows
+      if test.results.empty? # "no data after reset"
         expect do
-          table.read_rows.to_a # to_a is needed because Enum lazily invokes read_rows
-        end.must_raise Google::Cloud::Bigtable::InvalidRowStateError
+          rows.next
+        end.must_raise StopIteration
       else
-        rows = table.read_rows.to_a  # to_a is needed because Enum lazily invokes read_rows
-        rows.map(&:key).must_equal results.map(&:row_key).uniq
-
-        # The actual rows and cells in `rows` are nested, but the corresponding
-        # expected results are in a simple array, so flatten all cells in `rows`.
-        cells = rows.map do |row|
-          row.cells.values.flatten
-        end.flatten
+        cells = []
+        # Iterate over `results`, fetching rows and cells until each result has been used in assertions
+        test.results.each_with_index do |result|
+          if result.error
+            expect do
+              rows.next
+            end.must_raise Google::Cloud::Bigtable::InvalidRowStateError
+          else
+            if cells.empty?
```

Comment:
We don't have anything like this in the C# code, but I think that's just because our abstraction is "a stream of rows" so we flatten from those rows (via families and columns) to a sequence of cells, and then check all of those.

While this looks different, it looks like it's equivalent.